### PR TITLE
[FIX] point_of_sale: fix tax calculation when fiscal position selected to change taxes

### PR DESCRIPTION
**Pull request by Ephoniz**
Steps to reproduce:

- Install PoS, Accounting, and Sales
- Set up a product with taxes set as 15% and included in price.
- Now we set a fiscal position that changes the 15% included to 8% also included.
- If we test this in sales app it works properly.
- Now we go to PoS and select the product and the fiscal position.

Issue:

We can see that we have different results that if we do the same thing in sales, this is because the tax is calculated in pos based on the untaxed price from the 15% tax, and then we apply the fiscal position.

Solution:

Changed so we will calculate the tax based on the tax set in the fiscal position, assuming this is the expected behavior, and we don't want to modify the total price of the product, only how much of the price are taxes.

Reproductible until master

opw-3118888

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks good to me if we are nearer to what happen in sale.

@agr-odoo what do you think?

For a long time we did not support (or we supported but it was broken) to map to an included taxe. But I heard this could have changed recently.

**Comment by agr-odoo** [comment]
TL;DR Indeed currently we support included to included tax mapping and current behavior is fine

---

Since https://github.com/odoo/odoo/commit/625486a8382fde9033b73e2a4b4e7713cf4131c1 (and followup fixes) we support mapping an included tax to an included tax
The change was needed because `sale` and `account` didn't have the same behavior in this case and it was decided to use the `account` approach: keep the tax excluded price unit fixed

Simple example with numbers. Suppose we have:
* Product with price 121 and tax 21% included
* Fiscal position mapping the 21% included to 6% included 

The logic is that the sales customer benefit from a tax reduction, so the unit price, tax excluded, shall be the same (100) and the tax is reduced from 21 to 6, final price is thus 106. On the opposite, having the same amount total means the customer pays the same but there is an increase in unit price to compensate for the lower tax, and this would benefit the seller.
There are users who prefer the former approach and users who prefer the latter, so this will always generate some confusion, anyway the former is currently the default behavior in `sale`, `account` and `point_of_sale`

**The catch**

> But in sales order after changing the fiscal position the total price stay the same!

In `sales` and `account`, if you change the fiscal position *after* adding a line, the unit price will not change. If I remember correctly this is done partially to preserve old behavior but also because we avoid changing data already present in the document (unit price could be a manual input by the user)

**Comment by pierre-pde** [comment]
@agr-odoo , in your example, if you want to reach the price 106, wouldn't you configure that 6% of tax has 'tax excluded' ?

**Comment by agr-odoo** [comment]
@pierre-pde 
Yes, the output is the same. Conversely, to keep the total price 121 you could configure the source tax (21%) as excluded
The two approaches are sides of the same coin, `account` app behavior was identified as the correct one, so `sale` was adapted.

**Comment by dama-odoo** [comment]
Hi @Ephoniz, thanks for heading up and for suggesting fix, appreciated.
Just that I've hard time so far figuring out what's going wrong.

All the scenarii I've tested give the same figures on both Sales and PoS, whether the taxes behind the fiscal positions are included in the price or not.

Can you give an example (screenshot) of what you see ? Thanks !

**Comment by trj-odoo** [comment]
Just a little reminder, it would be good to add a more detailed description next time instead of "Try this in sales, and then try the same in PoS => result different". 
This forces us to guess the "strange behavior" depicted here unless we try it ourselves on runbot which is honestly a waste of time when having to review lot of PRs. As @agr-odoo explained, current behavior is fine. So this PR can be closed unless I am missing a point ?

**Comment by Ephoniz** [comment]
Ass discussed current behavior is ok.
